_docs/developer/testing/cypress.md

```diff
@@ -0,0 +1,187 @@
+
+```bash
+# from the site directory
+npx cypress run 
+```
+
+If any test fails a screen shot will be taken of the step that failed. Additionally, a video will be recorded each time of the run. They can be found under the `site/cypress/screenshots` and `site/cypress/videos` directories.
+
+See the full usage [here](https://docs.cypress.io/guides/tooling/plugins-guide.html#Use-Cases)
+
+## Writing Tests
+
+Test files are written under the `site/cypress/integration` and should follow the format `*.spec.js*` you can
+create folders as needed to organize groups of tests together.
+
+**Cypress test suite structure:**
+
+```
+cypress.json
+cypress/
+├── fixtures
+│   └── example.json
+├── integration
+│   └── ...
+│
+├── plugins
+│   └── index.js
+├── screenshots
+│ 
+├── support
+│   ├── commands.js
+│   └── index.js
+└── videos
+   
+```
+
+While the main test files are in the integration directory, there are some supporting files. The `fixtures` directory contains files used to mock data. This can be text files, JSON, images, etc. 
+
+The `commands.js` file 
+allow you to program Cypress commands. Actions that require multiple steps and might be done often can be grouped together under a single command to prevent repetition, for example the action of logging into the Submitty site is a command. You can also write plain JavaScript functions in this file and import them using ES6 module syntax.
+
+The `support/index.js` file instructs Cypress which files to include with every test, you most likely won't need to touch this file. 
+
+The `plugins/index.js` allows you to extend Cypress's default behavior. You can read more about them [here](https://docs.cypress.io/guides/tooling/plugins-guide.html#Use-Cases). 
+
+The top level `cypress.json` is the global configuration used.
+
+### Test Structure
+
+Cypress borrow's Mocha's Behavior-driven development (BDD) [test interface](https://mochajs.org/#bdd) to organize tests. 
+
+You will typically use `describes`, `context`, and `it` to organize tests. `describes` and `context` organize tests into groups, either or can be used they both act the same. Here is an example:
+
+```js
+describe('foo', () => {
+	context('bar', () => {
+		it('baz', () => {
+			expect(true).to.equal(true);
+		});
+	});
+});
+```
+
+This will produce the following structure in the test runner:
+
+![Cypress organization in the test runner](/images/cypress-structure.png)
+
+You can also use the `afterEach` and `beforeEach` function calls to perform setup and teardown for each test. 
+Be sure to read Cypress's best-practices for using these functions [here](https://docs.cypress.io/guides/references/best-practices.html#Using-after-or-afterEach-hooks).
+
+### Workflow 
+
+A test will typically:
+
+1. Navigate to a page
+2. Find an element
+3. *(Optionally) Interact with it*
+4. *(Optionally) Make an assertion*
```

Comment:
We should remove the `(Optionally)` bit here. Test cases that just go to a page and/or find an element should be the exception to the majority of pages that do interactions and make assertions.